.github/workflows/deploy-preview.yml

```diff
@@ -0,0 +1,194 @@
+name: deploy-preview
+
+on:
+  workflow_call:
+    inputs:
+      sha:
+        required: true
+        type: string
+      branch:
+        required: true
+        type: string
+      event_number:
+        required: true
+        type: string
+      title:
+        required: true
+        type: string
+      repo:
+        required: true
+        type: string
+      source_directory:
+        default: docs/sources
+        type: string
+      website_directory:
+        required: true
+        type: string
+      relative_prefix:
+        required: true
+        type: string
+
+env:
+  CLOUD_RUN_REGION: us-south1
+
+permissions:
+  contents: read
+  id-token: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ inputs.repo }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  deploy-preview:
+    permissions: write-all
```

Comment:
I know it may be a pain to whittle down the list but do we really need all the write permissions?

From my reading we need `id-token: write` for the auth parts, and `statuses: write` for the commit status updates, and `pull-requests: write` to put the comments.

If we do list the permissions explicitly, it's probably worth commenting what purpose each permission serves perhaps with a link or mention of the step that needs it so we can remember in the future.